cdap-api/src/main/java/co/cask/cdap/api/app/ApplicationContext.java

```diff
@@ -30,7 +32,8 @@
   /**
    * Get the configuration object.
    *
-   * @return configuration object
+   * @return application configuration provided during application creation, else null
```

Comment:
I think it would be better to return an empty Config object rather than returning null. Otherwise every app needs to perform null checks before using the config.